libsolidity/interface/StandardCompiler.cpp

```diff
@@ -1647,7 +1647,7 @@ Json StandardCompiler::compileYul(InputsAndSettings _inputsAndSettings)
 	std::string const& sourceContents = _inputsAndSettings.sources.begin()->second;
 
 	// Inconsistent state - stop here to receive error reports from users
-	if (!stack.parseAndAnalyze(sourceName, sourceContents) && stack.errors().empty())
+	if (!stack.parseAndAnalyze(sourceName, sourceContents) && !stack.hasErrors())
```

Comment:
Although there are no warnings in Yul Parser, if that happens to be the case in the future, this would already prevent the mistake of consider them as errors. Also, it follows the same logic of the changes made to instances of `ErrorReporter::errors()`.

Reply:
I mean, we have no warnings in `AsmParser` but we do have some in `AsmAnalysis`. So you can actually get a warning from `parseAndAnalyze()`.

But anyway, this is a good change.

Reply:
Actually, I'd go even further and make it an assert. That's what we do already in `compileSolidity()`. The exception will then be caught by the top-level handler in `StandardCompiler::compile()` and reported as error anyway.

This will affect the behavior slightly - the user will no longer get anything already added to `output` - but I think that's the correct behavior in case of a failed assumption anyway. I.e. drop everything and just fail immediately.

Reply:
I see that you changed it to `!Error::hasErrorsWarningsOrInfos(stack.errors())`, but like I explained, warnings are actually possible so we want `!hasErrors()` instead. When the result is `false` but you don't get any errors, only warnings, something is wrong.

Reply:
Fixed.